### PR TITLE
[deployer] - Impl Synnax Deployer, resolves #355

### LINKED ISSUE #355: [SY-190] Automate Deployment on GSE DAQ
Now that we have the new GSE DAQ PC, I'd like to automate the deployment of new releases. This issue will require a good amount of research. Use docker compose? Use k8s? k3s?

<sub>From [SyncLinear.com](https://synclinear.com) | [SY-190](https://linear.app/synnaxlabs/issue/SY-190/automate-deployment-on-gse-daq)</sub>

### PR DESCRIPTION
**Pull request by MatchaOnMuffins**
This PR addresses issue #355 by providing a Docker-based lifecycle management solution to synnax.

### CODE REVIEW

File: deploy/deployment.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import docker
+import requests
+import time
+import logging
+
+# TODO: expose these as environment variables
+
+image_name = "synnaxlabs/synnax"  # image name
```
Comment (emilbon99) [inline deploy/deployment.py:8]: Can we make these constants SCREAM_CASE?

File: deploy/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+docker
```
Comment (emilbon99) [inline deploy/requirements.txt:1]: Would it be possible to use poetry here for the virtual environment? Just want to make sure we're standardized across the repo on these things.

**Review by emilbon99 — CHANGES_REQUESTED** [review]
Two issues to think about.

**Comment by MatchaOnMuffins** [comment]
Just resolved the two issues you mentioned. Can you take a look gain?

**Comment by emilbon99** [comment]
Hey, sorry I didn't get to this today. Will get to it tomorrow

**Pull request merged**